src/Automatonymous/States/StateMachineState.cs

```diff
@@ -216,9 +214,15 @@ public void AddSubstate(State<TInstance> subState)
             if (_name.Equals(subState.Name))
                 throw new ArgumentException("A state cannot be a substate of itself", nameof(subState));
 
+            subState.SetSuperState(this);
             _subStates.Add(subState);
         }
 
+        public void SetSuperState(State<TInstance> superState)

```

Comment:
Yeah, I still don't see a reason for this to be modifiable but there are a lot of changes. Also, this would be another internal method I think.

Reply:
I agree. It felt like a hack at the time. It has never felt right. I think based on the decision above (about how to declare SubState) will change whether I can remove it altogether or simply make it internal. Let me know your thoughts on that previous comment.

Reply:
@phatboyg Removed this for now. The builder method `SubState(State<TInstance>, State<TInstance>)` was the method that required it. I removed this. I can see this causing inconsistencies anyways.